src/sentry/hybridcloud/rpc/service.py

```diff
@@ -502,10 +501,10 @@ def address(self) -> str:
 
     @property
     def path(self) -> str:
-        return django.urls.reverse(
-            "sentry-api-0-rpc-service",
-            kwargs={"service_name": self.service_name, "method_name": self.method_name},
-        )
+        # don't use reverse here because this is an extremely hot path
+        # and we kept seeing perf issues to do with lock contention
+        # with how django's reverse is implemented
+        return f"/api/0/internal/rpc/{self.service_name}/{self.method_name}/"
```

Comment:
we should be careful here that we're not substituting things that would need url escaping (that or we should call urllib.parse.quote on these) -- these are probably always from a fixed set of strings (maybe `LiteralString` could help enforce this at typing time?)

Reply:
There shouldn't be an URL unsafe data here because both the service name and method have to be python symbol names. Applying `urllib.parse.quote` would be a good backstop though.